### PR TITLE
fix/ remove k8s volumes from TMHOME

### PR DESCRIPTION
**Pull request by 0xArdi**
## Fixes

This PR copies the config of a tendermint node from the shared volume to the local fs of the container it is being run on, when using k8s builds.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [x] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: autonomy/deploy/generators/kubernetes/templates.py [hunk 1]
```diff
@@ -281,6 +281,17 @@
         - name: nodes
           persistentVolumeClaim:
             claimName: 'nodes'
+        - emptyDir: {{}}
+          name: local-tendermint
+      initContainers:
+        - name: copy-tendermint-configuration
+          image: "ubuntu/ubuntu:20.04"
+          command: ["cp", "-r", "/tendermint/node{validator_ix}", "/tm"]
```
Comment (DavidMinarsch) [inline autonomy/deploy/generators/kubernetes/templates.py:289]: this doesn't match above?

Comment (8ball030) [reply autonomy/deploy/generators/kubernetes/templates.py]: We are doing a bit of juggling in the init container.

init container starts with 2 volume mounts;
local-tendermint -> this is a host path pv.
nodes -> where the config get written to.

We copy the directory when it is available, and, additionally we dont copy over all of the configs, only the oone relevant for our use.

File: autonomy/deploy/generators/kubernetes/templates.py [hunk 2]
```diff
@@ -281,6 +281,19 @@
...
           persistentVolumeClaim:
             claimName: 'nodes'
+        - emptyDir: {{}}
+          name: local-tendermint
+      initContainers:
+        - name: copy-tendermint-configuration
+          image: "ubuntu:20.04"
+          command: ["bash", "-c"]
+          args: 
+          - "while [ ! -d /tendermint/node{validator_ix} ]; do sleep 1; done; cp -r /tendermint/node{validator_ix}/* /tm/"
```
Comment (8ball030) [inline autonomy/deploy/generators/kubernetes/templates.py:291]: wait until the file is available, otherwise we instantly complete initialisation.

Comment (0xArdi) [reply autonomy/deploy/generators/kubernetes/templates.py:291]: @angrybayblade i think this is just explaining what is going on. @8ball030 can confirm.

**Review by angrybayblade — APPROVED** [review]

**Pull request merged**